runtime/stdlib/contracts/test.cdc

```diff
@@ -0,0 +1,236 @@
+        /// Returns a new matcher that succeeds if this and the given matcher succeed.
+        ///
+        pub fun or(_ other: Matcher): Matcher {
+            return Matcher(test: fun (value: AnyStruct): Bool {
+                return self.test(value) || other.test(value)
+            })
+        }
+    }
+
+    /// ResultStatus indicates status of a transaction or script execution.
+    ///
+    pub enum ResultStatus: UInt8 {
+        pub case succeeded
+        pub case failed
+    }
+
+    /// The result of a transaction execution.
+    ///
+    pub struct TransactionResult {
+        pub let status: ResultStatus
+        pub let error: Error?
+
+        init(status: ResultStatus, error: Error) {
+            self.status = status
+            self.error = error
+        }
+    }
+
+    /// The result of a script execution.
+    ///
+    pub struct ScriptResult {
+        pub let status: ResultStatus
+        pub let returnValue: AnyStruct?
+        pub let error: Error?
+
+        init(status: ResultStatus, returnValue: AnyStruct?, error: Error?) {
+            self.status = status
+            self.returnValue = returnValue
+            self.error = error
+        }
+    }
+
+    // Error is returned if something has gone wrong.
+    //
+    pub struct Error {
+        pub let message: String
+
+        init(_ message: String) {
+            self.message = message
+        }
+    }
+
+    /// Account represents info about the account created on the blockchain.
+    ///
+    pub struct Account {
+        pub let address: Address
+        pub let publicKey: PublicKey
+
+        init(address: Address, publicKey: PublicKey) {
+            self.address = address
+            self.publicKey = publicKey
+        }
+    }
+
+    /// Configuration to be used by the blockchain.
+    /// Can be used to set the address mappings.
+    ///
+    pub struct Configuration {
+        pub let addresses: {String: Address}
+
+        init(addresses: {String: Address}) {
+            self.addresses = addresses
+        }
+    }
+
+    /// Transaction that can be submitted and executed on the blockchain.
+    ///
+    pub struct Transaction {
+        pub let code: String
+        pub let authorizers: [Address]
+        pub let signers: [Account]
+        pub let arguments: [AnyStruct]
+
+        init(code: String, authorizers: [Address], signers: [Account], arguments: [AnyStruct]) {
+            self.code = code
+            self.authorizers = authorizers
+            self.signers = signers
+            self.arguments = arguments
+        }
+    }
+
+    /// BlockchainBackend is the interface to be implemented by the backend providers.
+    ///
+    pub struct interface BlockchainBackend {
+
+        pub fun executeScript(_ script: String, _ arguments: [AnyStruct]): ScriptResult
+
+        pub fun createAccount(): Account
+
+        pub fun addTransaction(_ transaction: Transaction)
+
+        pub fun executeNextTransaction(): TransactionResult?
```

Comment:
I think this needs documentation about the specific circumstances in which this can return `nil`.

Reply:
It is mentioned in the corresponding public/wrapper method (https://github.com/onflow/cadence/blob/4fd33f59f6af3cbc11948606ef3c4cab6ca0b471/runtime/stdlib/contracts/test.cdc#L43-L48). I can copy over the same here.